Lib/configparser.py

```diff
@@ -279,6 +279,13 @@ class InterpolationSyntaxError(InterpolationError):
     which substitutions are made does not conform to the required syntax.
     """
 
+class InterpolationTypeError(Error):
```

Comment:
If you add a new exception type then you need to add it to the documentation as well. TBH, I don't think you need a new exception type here. You could just raise a TypeError.